src/Components/allocations/helper_functions/getAllocations.ts

```diff
@@ -0,0 +1,20 @@
+import { Dispatch, SetStateAction } from "react";
+import { loadDataFromFirebase } from "../../../dataBaseUtils/readWrite";
+import { allocationType } from "../../../types";
+
+function getAllocations(setAllocations: Dispatch<SetStateAction<allocationType[]>>) {
```

Comment:
Same consideration as in `deleteAllocations`. You don't want to pass setter into such a function. You can still handle errors here, but just return the result and then use this result as you with in the caller.